### PR TITLE
Add solar API documentation

### PR DESCRIPTION
**Pull request by SanJoseBart**
Adding documentation for a couple of the solar API endpoints: `/api/1/energy_sites/{site_id}/history` and `/api/1/energy_sites/{site_id}/calendar_history`.  I trust this repository is appropriate for solar API documentation in addition to vehicle API documentation; it seems more current and active than other locations I've searched.

This is my first usage of GitHub, so please let me know of any conventions I missed.

### CODE REVIEW

**Comment by core-hacked** [comment]
Although the documentation of the endpoint looks correct, it is a bit unclean and needs some workups.
You can look at how past documentation has been written and see how you could improve this.

Kind Regards!

**Comment by SanJoseBart** [comment]
I did read the other API specs and I did try to match the formatting.  However, I also separated the two usages of each endpoint into two separate APIs, which I thought was clearer.  And I did remove the JSON color-coding so I could include a "..." to indicate multiple array entries.  I've now edited the history.md file to remove those two differences.  How does this one look?

Cheers!

**Comment by core-hacked** [comment]
@SanJoseBart This one looks better, I would still remove the mentions of the owner-api uri, and the oauth bearer token, as the usage is already implied by the previous documentation. I would just describe the endpoints, what they do and how to call them. Look at how the vehicle commands with parameters are structured. I wouldn't split the documentation like that for an endpoint on the same request URI.

Although you can also wait on what Tim Dorr (the owner) says, he probably shares the same or similar opinion.

**Comment by SanJoseBart** [comment]
I removed the authentication notes, and moved the products API to its own file.  However, I have to disagree with "just describe the endpoints, what they do and how to call them".  We should also  document the responses for APIs whose purpose is to retrieve information.  I can see omitting the responses for the vehicle commands, but that's a different situation.

I combined the request parameter tables for the two use cases into a single table to help with the "split" appearance, but I left both responses.  Hopefully that will be okay.

File: docs/solar/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+description: Solar API Overview
+---
+
+# Solar API Endpoints
+
+The base URI for all site-specific solar APIs is `https://owner-api.teslamotors.com/api/1/energy_sites/{site_id}/` followed by the endpoint name and parameters. `site_id` identifies the solar installation.  See the [products.md](products.md) file for obtaining the `site_id` value.
```
Comment (timdorr) [inline docs/solar/README.md:7]:
```suggestion
The base URI for all site-specific solar APIs is `https://owner-api.teslamotors.com/api/1/energy_sites/{site_id}/` followed by the endpoint name and parameters. `site_id` identifies the solar installation.  See the [`products` endpoint](products.md) for obtaining the `site_id` value.
```

You should link to logical titles/names of documents, not their files. Most users aren't browsing the repository, they're reading the Gitbook site.

File: docs/solar/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+The solar endpoints fall into three groups:
+
+### Solar energy generation data endpoints
+
+| Cmd | Endpoint            | Description                                    |
+| :-- | :------------------ | :--------------------------------------------- |
+| GET | `history`           | Current and immediately prior generation data. |
+| GET | `calendar_history`  | Generation data from a past period.            |
+| GET | `telemetry_history` | No information available.                      |
```
Comment (timdorr) [inline docs/solar/README.md:17]: We don't need to put these endpoints as a table of contents list. You can use Gitbook to link to them. 

To change the main sidebar, you use the Summary file: https://github.com/timdorr/tesla-api/blob/master/docs/SUMMARY.md

File: docs/solar/products.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
```
Comment (timdorr) [inline docs/solar/products.md:1]: This file should go under `api-basics`, since it's a top-level endpoint.

**Comment by timdorr** [comment]
OK, I cleaned this up and formatted it to better integrate on the Gitbook site. Thanks for getting this going and making my life easy, @SanJoseBart!

**Pull request merged**